src/delegate/client.rs

```diff
@@ -23,17 +23,21 @@ use super::not_initialized_error;
 /// Maps all pending client request IDs to their future responses.
 type RequestMap = DashMap<u64, oneshot::Sender<Output>>;
 
-/// Handle for communicating with the language client.
+/// Internals for language client handle.
 #[derive(Debug)]
-pub struct Client {
+pub(super) struct ClientInner {
     sender: Sender<String>,
     initialized: Arc<AtomicBool>,
     request_id: AtomicU64,
     pending_requests: Arc<RequestMap>,
 }
 
-impl Client {
-    pub(super) fn new(
+/// Handle for communicating with the language client.
+#[derive(Clone, Debug)]
+pub struct Client(pub(super) Arc<ClientInner>);
```

Comment:
I don't think this field needs to be `pub(super)` either. Mind if you could change this to be private?

Reply:
Good catch. I thought those were needed for some reason to get it to compile but you are right.